base-R.qmd

```diff
@@ -0,0 +1,502 @@
+# A field guide to base R
+
+```{r}
+#| results: "asis"
+#| echo: false
+source("_common.R")
+status("complete")
+```
+
+To finish off the programming section, we're going to give you a quick tour of the most important base R functions that we don't otherwise discuss in the book.
+This will help you read code that you'll encounter in the wild and start your on your journey to learn different ways to tackle the same problems.
+
+We'll focus on three big topics: subsetting with `[`, subsetting with `[[` and `$` and iteration with for loops and the apply family of functions.
+These tools are particularly useful as you being to do more programming in R, which is why we don't teach them earlier.
+
+This is a good place to remind you that the tidyverse is not only way to solve data science problems.
+We teach the tidyverse in this book because it's designed as whole, which tends to increase consistency across functions making it a little easier to use.
+After you read this book we expect you'll learn other ways to tackle the same problems using base R, data.table, and other packages.
+You'll certainly encounter these other approaches when you start reading R code written by other people, particularly if you're using StackOverflow.
+It's 100% OK for your code to include a mix of paradigms, but when you have time, we highly recommend taking the time to rewrite with a consistent style --- this will help you understand what it does better, improve your R programming skills, and lead to code that's easier to maintain.
+
```

Comment:
Maybe some reminder that you've really been using base R all along throughout this book. The entire `Transform` section uses data types and functions that come from base R, i.e. we didn't invent factors/Date objects, and we didn't invent the `&` or `>` functions.